[Doman Purchase] Site Preview UI

alpavanoglu opened the pull request:
Fixes #20080 

## Testing Steps

### Domain Purchase Flag Off
1. Install & Login Jetpack
2. Tap on "Create WordPress.com site" from My Sites menu.
3. Complete the flow with a free domain.
4. The Preview UI should look the same as before aside from title being left (natural) aligned.

### Domain Purchase Flag On & Free Domain
1. Install & Login Jetpack
2. Tap on "Create WordPress.com site" from My Sites menu.
3. Complete the flow with a **free** domain.
4. The Preview UI should match the new designs but the description and footer below should not be visible as those texts are only for Paid Domains. Title should also be as before "Your site has been created".

### Domain Purchase Flag On & Paid Domain
1. Install & Login Jetpack
2. Tap on "Create WordPress.com site" from My Sites menu.
3. Complete the flow with a **paid** domain.
4. The Preview UI should match the new designs with a description and a footer. Verify the strings as well.

## Regression Notes
1. Potential unintended areas of impact
N/A

7. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

8. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

alpavanoglu committed:
Add Domain Purchase specific labels to `SiteAssemblyContentView`

alpavanoglu committed:
Update strings for free & paid domains

salimbraksa commented on WordPress/Classes/ViewRelated/Site Creation/Final Assembly/SiteAssemblyContentView.swift:
```diff
@@ -1,4 +1,4 @@
-
+import Foundation
 import UIKit
```
nit: no need to import Foundation if we're already importing UIKit.

salimbraksa commented on WordPress/Classes/ViewRelated/Site Creation/Final Assembly/SiteAssemblyContentView.swift:
```diff
@@ -104,12 +140,16 @@ final class SiteAssemblyContentView: UIView {
     }
 
     let siteCreator: SiteCreator
+    private lazy var shouldShowDomainPurchase: Bool = {
```
nit: perhaps there should be a `\n` between `siteCreator` property and `shouldShowDomainPurchase`.

alpavanoglu committed:
Merge branch 'trunk' into feature/domain-purchase-preview-ui

alpavanoglu committed:
Update broken unit test

alpavanoglu committed:
Fix `SiteAssemblyViewTests`

alpavanoglu committed:
Merge branch 'trunk' into feature/domain-purchase-preview-ui

alpavanoglu committed:
Merge branch 'trunk' into feature/domain-purchase-preview-ui

mokagio committed:
Track result of running `bundle exec pod install`

Sometimes, an out of date local cache can result in `FBReactNativeSpec`
changing even when it shouldn't. That must be what happened on the
branch we're working here, as shown by this diff and the CI failures at
https://buildkite.com/automattic/wordpress-ios/builds/13717#_

mokagio commented:
I've enabled auto-merge and fixed the corrupted `Podfile.lock` that [CI picked up](https://buildkite.com/automattic/wordpress-ios/builds/13717#_) to get ahead with the preparation for the upcoming 22.2 code freeze.
